python_modules/dagster/dagster/_core/remote_representation/code_location.py

```diff
@@ -826,47 +841,56 @@ def get_subset_external_job_result(
     def get_external_partition_config(
         self,
         repository_handle: RepositoryHandle,
-        partition_set_name: str,
+        job_name: str,
         partition_name: str,
         instance: DagsterInstance,
     ) -> "ExternalPartitionConfigData":
         check.inst_param(repository_handle, "repository_handle", RepositoryHandle)
-        check.str_param(partition_set_name, "partition_set_name")
+        check.str_param(job_name, "job_name")
         check.str_param(partition_name, "partition_name")
 
         return sync_get_external_partition_config_grpc(
-            self.client, repository_handle, partition_set_name, partition_name, instance
+            self.client, repository_handle, job_name, partition_name, instance
         )
 
     def get_external_partition_tags(
         self,
         repository_handle: RepositoryHandle,
-        partition_set_name: str,
+        job_name: str,
         partition_name: str,
         instance: DagsterInstance,
+        selected_asset_keys: Optional[AbstractSet[AssetKey]],
     ) -> "ExternalPartitionTagsData":
         check.inst_param(repository_handle, "repository_handle", RepositoryHandle)
-        check.str_param(partition_set_name, "partition_set_name")
+        check.str_param(job_name, "job_name")
         check.str_param(partition_name, "partition_name")
 
         return sync_get_external_partition_tags_grpc(
-            self.client, repository_handle, partition_set_name, partition_name, instance
+            self.client, repository_handle, job_name, partition_name, instance, selected_asset_keys
         )
 
     def get_external_partition_names(
-        self, external_partition_set: ExternalPartitionSet, instance: DagsterInstance
+        self,
+        repository_handle: RepositoryHandle,
+        job_name: str,
+        instance: DagsterInstance,
+        selected_asset_keys: Optional[AbstractSet[AssetKey]],
     ) -> "ExternalPartitionNamesData":
-        check.inst_param(external_partition_set, "external_partition_set", ExternalPartitionSet)
+        external_repo = self.get_repository(repository_handle.repository_name)
+        partition_set_name = external_partition_set_name_for_job_name(job_name)
 
-        # Prefer to return the names without calling out to user code if the
-        # partition set allows it
-        if external_partition_set.has_partition_name_data():
-            return ExternalPartitionNamesData(
-                partition_names=external_partition_set.get_partition_names(instance=instance)
-            )
+        # Prefer to return the names without calling out to user code if there's a corresponding
+        # partition set that allows it
```

Comment:
ah OK, we are still keeping the optimization mentioned above in the place where it might actually matter, cool